src/tds/query.c

```diff
@@ -728,12 +728,48 @@ tds_get_column_declaration(TDSSOCKET * tds, TDSCOLUMN * curcol, char *out)
 	size = tds_fix_column_size(tds, curcol);
 
 	switch (tds_get_conversion_type(curcol->on_server.column_type, curcol->on_server.column_size)) {
-	case XSYBCHAR:
+	case XSYBCHAR: //same as SYBLONGCHAR
+		if (!IS_TDS7_PLUS(tds->conn)) { //SYBLONGCHAR
+			switch (curcol->column_usertype) { // dirty temporary hack
+			case 2:
+				fmt = "VARCHAR(%u)";
+				break;
+			case 24:
+				fmt = "NCHAR(%u)";
+				break;
+			case 25:
+				fmt = "NVARCHAR(%u)";
+				break;
+			case 1:
+			default:
+				fmt = "CHAR(%u)";
+				break;
+			}
+			max_len = 32767;
+			break;
+		}
 	case SYBCHAR:
-		fmt = "CHAR(%u)";
+		if (!IS_TDS7_PLUS(tds->conn)) {
+			switch (curcol->column_usertype) { // dirty temporary hack
+			case 24:
+				fmt = "NCHAR(%u)";
+				break;
+			case 1:				
+			default:
+				fmt = "CHAR(%u)";
+				break;
+			}
+		}
 		break;
 	case SYBVARCHAR:
-	case XSYBVARCHAR:
+		if (!IS_TDS7_PLUS(tds->conn) && curcol->column_usertype == 25) {
+			if (curcol->column_varint_size == 8)
+				fmt = "NVARCHAR(MAX)";
```

Comment:
I don't think Sybase supports `nvarchar(max)`

Reply:
To be honest I haven't tested that part of the code, and just replicated similar portion. It's that a case which can't occur (col varint size = 8), I can remove it. Or I can default to 32767?

Reply:
Sybase does not support `varchar(max)` or, which is similar (there's also `varbinary(max)`), `column_varint_size` == 8.

Reply:
Indeed. Sybase (var)char max length is 16384 (or apparently 16296 depending on pagesize and table lock scheme). I'm not quite sure how to handle this. 

My little book says that this value can be retrieved with **dbcc serverlimits** but that would give a database level limit... also, not everyone can execute **dbcc**.

Is that information carried over by TDS by any chance to your knowledge? I would be surprised if it is.

Should I just stick to 16296?